index.rb

```diff
@@ -1,160 +1,234 @@
-        my_all_boolean = false
+    # Checks for when a block is not given
+    unless block_given?
+      my_all_boolean = true
+      each do |i|
+        my_all_boolean = false unless i
       end
+      return my_all_boolean
+    end
 
+    # If a block is given
+    my_all_boolean = true
+    each do |i|
+      my_all_boolean = false unless yield(i)
     end
-    puts my_all_boolean
+    my_all_boolean
   end
 
-  def my_any?
-    if self.length == 0 then
-      return false
+  def my_any?(*args)
+    if args.length == 1 && !block_given?
+
+      if args.first.instance_of?(Regexp)
+        my_any_boolean = false
+        each do |i|
+          my_any_boolean = true if args.first.match(i)
+        end
+        return my_any_boolean
+      end
+
+      if args.first.instance_of?(Class)
+        my_any_boolean = false
+        each do |i|
+          my_any_boolean = true if i.is_a?(args.first)
+        end
+        return my_any_boolean
+      end
+
+      my_any_boolean = false
+      each do |i|
+        my_any_boolean = true if i == args.first
+      end
+      return my_any_boolean
+    end
+    unless block_given?
+      my_any_boolean = false
+      each do |i|
+        my_any_boolean = true if i
+      end
+      return my_any_boolean
     end
     my_any_boolean = false
+    each do |i|
+      my_any_boolean = true if yield(i)
+    end
+    my_any_boolean
+  end
 
-    for i in self
+  def my_none?(*args)
+    if args.length == 1 && !block_given?
 
-      if yield(i)
-        my_any_boolean = true
+      if args.first.instance_of?(Regexp)
+        my_none_boolean = true
+        each do |i|
+          my_none_boolean = false if args.first.match(i)
+        end
+        return my_none_boolean
       end
 
+      my_none_boolean = true
+      each do |i|
+        my_none_boolean = false if i == args.first
+      end
+      return my_none_boolean
     end
-    puts my_any_boolean
-  end
 
-  def my_none?
-    if self.length == 0 then
-      return true
+    unless block_given?
+      my_none_boolean = true
+      each do |i|
+        my_none_boolean = false if i
+      end
+      return my_none_boolean
     end
-    my_none_boolean = true
-    for i in self
 
-      if yield(i)
-        my_none_boolean = false
+    if args.length == 1
+      my_none_boolean = true
+      each do |i|
+        my_none_boolean = false if i.is_a?(args.first)
       end
+      return my_none_boolean
 
     end
-    puts my_none_boolean
+
+    my_none_boolean = true
+    each do |i|
+      my_none_boolean = false if yield(i)
+    end
+    my_none_boolean
+  end
+
+  def my_inject(memo = nil, sym = nil, &block)
+    memo = memo.to_sym if memo.is_a?(String) && !sym && !block
+    if memo.is_a?(Symbol) && !sym
+      block = memo.to_proc
+      memo = nil
+    end
+    sym = sym.to_sym if sym.is_a?(String)
+    block = sym.to_proc if sym.is_a?(Symbol)
+
+    # Ready to rock & roll
+    each { |x| memo = memo.nil? ? x : block.yield(memo, x) }
+    memo
   end
```

Comment:
#my_inject raises a "LocalJumpError" when no block or argument is given.
Expected localJumpError. instead, got NoMethodError: undefined method  'yield'. You need to refactor the code to correctly use yield.
Using the regular built-in enumerable method like **each** is not allowed. We are supposed to create methods that mimic them.